staging/src/k8s.io/client-go/dynamic/client_pool.go

```diff
@@ -70,6 +71,13 @@ func NewClientPool(config *restclient.Config, mapper meta.RESTMapper, apiPathRes
 	}
 }
 
+// Instantiates a new dynamic client pool with the given config.
+func NewDynamicClientPool(cfg *restclient.Config) ClientPool {
```

Comment:
This is causing `hack/verify-staging-imports.sh` to fail with:
```
I0131 11:53:14.800] client-go has a forbidden dependency:
I0131 11:53:14.800] 
I0131 11:53:14.802]   k8s.io/client-go/dynamic imports k8s.io/kubernetes/pkg/api
I0131 11:53:14.802] 
I0131 11:53:14.803] FAILED   hack/make-rules/../../hack/verify-staging-imports.sh
```

So I will have to move this elsewhere.
I had it at `pkg/controller/client` earlier.
@derekwaynecarr Is that fine or is there some other place you want me to put this at?